lib/cstruct.ml

```diff
@@ -96,7 +96,7 @@ let create_unsafe len =
   { buffer ; len ; off = 0 }
 
 let check_bounds t len =
-  Bigarray.Array1.dim t.buffer >= len
+  len >= 0 && Bigarray.Array1.dim t.buffer >= len
```

Comment:
> It'd be quite handy to have arithmetic operations that raise on overflow in the standard library.

the base for this was already done in https://github.com/ocaml/ocaml/pull/1060 (merged in trunk, not part of 4.05) -- but these C symbols (esp. add/sub) are not yet used anywhere.